msteams-platform/toolkit/debug-your-Teams-app-test-tool.md

```diff
@@ -0,0 +1,288 @@
+
+1. **Integration with existing Teams bot applications**: The Test Tool can be integrated with existing Teams bot applications that are built with [Bot Framework SDK](https://dev.botframework.com/).
+
+1. **Support for different scopes**: The Test Tool supports testing in personal, team, and group chat scopes.
+
+### Limitations
+
+The following are the limitations:
+
+1. The Test Tool doesn't process the app manifest, which means features that are only enabled through the manifest won't be accessible.
+
+1. The Test Tool doesn't support all other types of Bot application [Cards](../task-modules-and-cards/what-are-cards.md#cards).
+
+1. The Test Tool doesn't support the following Adaptive Card features:
+
+   * [People Picker](../task-modules-and-cards/cards/people-picker.md)
+   * [Typeahead search](../task-modules-and-cards/cards/dynamic-search.md)
+   * [User mention](../task-modules-and-cards/cards/cards-format.md#microsoft-azure-active-directory-azure-ad-object-id-and-upn-in-user-mention)
+   * [Stage View](../task-modules-and-cards/cards/cards-format.md#stage-view-for-images-in-adaptive-cards)
+   * [Full width](../task-modules-and-cards/cards/cards-format.md#full-width-adaptive-card)
+
+1. The Test Tool doesn't support the following:
+   * Single sign-on (SSO) 
+   * Mobile 
+   * Meetings 
+
+1. **Limited emulation**
+
+   | Features | Debug in Test Tool | Local debug |
+   | --- | --- | --- |
+   | Basic sending / receiving messages | Available | Available |
+   | Bot Framework APIs (TeamsInfo.getPagedMembers()...) |Available (respond with mocked data) | Available |
+   | Sending Teams events | Available (mocking activity) |Available |
+   | Typing indicator | Not Available | Available |
+   | Tab, Messaging Extension, Task Module, SSO, and non-Adaptive Cards | Not Available | Available |
+
+### Teams Toolkit samples and templates
+
+Test the bot sample, which displays the features supported by the Teams App Test Tool, as follows:
+
+1. Open [bot sample](https://github.com/OfficeDev/TeamsFx-Samples/tree/v3).
+
+1. Select **Code**.
+
+1. From the dropdown menu, select **Open with GitHub Desktop**.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug/test-tool-clone-repository.png" alt-text="Screenshot show the option to clone repository in local.":::
+
+1. Select **Clone**.
+
+1. Open the project folder in Visual Studio Code with Teams Toolkit.
+
+1. From the left pane, select **Run and Debug** (`Ctrl+Shift+D`) and select **Debug in Test Tool** in dropdown list.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug/select-debug-in-test-tool.png" alt-text="Screenshot shows the option to select debug in test tool.":::
+
+1. Select **F5**.
+
+1. Type **help** command. 
+
+A webpage opens with a list of commands.
+
+   :::image type="content" source="../assets/images/teams-toolkit-v2/debug/test-tool.png" alt-text="Screenshot shows the Teams App Test Tool Adaptive Card.":::
```

Comment:
The image looks blurry, would you mind to replace it with a clear version?

Reply:
Lightbox added.